data_source/gateway.go

```diff
@@ -27,14 +27,144 @@ func (g *Gateway) GetBlockByNumber(blockNumber uint64) (*core.Block, error) {
 
 // GetTransaction gets the transaction for a given transaction hash from the feeder gateway,
 // then adapts it to the appropriate core.Transaction types.
-func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (*core.Transaction, error) {
-	return nil, errors.New("not implemented")
+func (g *Gateway) GetTransaction(transactionHash *felt.Felt) (core.Transaction, error) {
+	response, err := g.client.GetTransaction(transactionHash)
+	if err != nil {
+		return nil, err
+	}
+
+	tx, err := adaptTransaction(response, g)
+	if err != nil {
+		return nil, err
+	}
+
+	return tx, nil
+}
+
+func adaptTransaction(response *clients.TransactionStatus, g *Gateway) (core.Transaction, error) {
```

Comment:
Instead of passing in the transaction status, I think it's better to pass in the transaction instead. 
```suggestion
func adaptTransaction(transaction *clients.Transaction, g *Gateway) (core.Transaction, error) {
```